toffy/streak_detection_test.py

```diff
@@ -0,0 +1,178 @@
+                    for init_tiff in Path(self.data_dir / fov.fov_name).glob("*.tiff")
+                ]
+
+                # Test that the set of initial tiffs map 1-to-1 to set of corrected tiffs
+                assert Counter(initial_tiff_names) == Counter(corrected_tiff_names)
+
+
+# ? Construct different sets of fake data (noisy, and clean variants)
+# ? Modify to work with `parameterize_with_cases`
+# TODO: All channels of fake data must contain the same streaks. Add noise to each channel.
+# TODO: Add multiple streaks.
+# TODO: Hardcode the streak tests first, request for review
+@pytest.fixture(scope="class", params=[2, 4, 10])
+def fake_data(request):
+    # Set up fake data dimensions
+    num_images = 10
+    row_size = 1000
+    col_size = 1000
+    fake_data = np.zeros(shape=(row_size, col_size, num_images), dtype=np.uint8)
+
+    # Initialize a new generator
+    rng = np.random.default_rng(12345)
+
+    # Generate Streaks
+    for _ in range(request.param):
+        (x_min,) = rng.integers(low=0, high=col_size - 50, size=1)
+        (x_max,) = rng.integers(low=x_min + 50, high=col_size - 1, size=1)
+        (y,) = rng.integers(low=0, high=row_size - 1, size=1)
+        fake_streak = np.ones(shape=(x_max - x_min, num_images))
+        fake_data[y, x_min:x_max, :] = fake_streak
+
+    # for fd in range(num_images):
+
+    # Create fake DataArray
+    fake_data_da = xr.DataArray(
+        data=fake_data,
+        coords=[range(row_size), range(col_size), range(num_images)],
+        dims=["rows", "cols", "channels"],
+    )
+    # Create fake StreakData dataclass
+    # Use channel 0 as the streak detection channel.
+    streak_data: sd.StreakData = sd.StreakData(
+        shape=(row_size, col_size), fov="fake_fov", streak_channel=0
+    )
+
+    # Add the Fake Fov to the request object
+    request.cls.fake_fov = Fov(
+        fov_name="fake_fov", fov_da=fake_data_da, corrected_channels=None, streak_data=streak_data
+    )
+
+
+@pytest.mark.usefixtures("fake_data")
+class TestHelperFunctions:
+    def test_make_binary_mask(self):
+
+        for channel in self.fake_fov.fov_da.channels.values:
+            fake_img_bin_mask: np.ndarray = sd._make_binary_mask(
+                input_image=self.fake_fov.fov_da.loc[:, :, channel].values
+            )
+            # Make sure the masks only contain values: `0`, `1`
+            assert np.all(np.isin(fake_img_bin_mask, [0, 1]))
+
+            # Make sure the shape is the same
+            assert fake_img_bin_mask.shape == self.fake_fov.streak_data.shape
+
+    @pytest.mark.parametrize("min_length", [50])
+    def test_make_mask_dataframe(self, min_length):
+
+        fake_img_bin_mask = sd._make_binary_mask(
+            input_image=self.fake_fov.fov_da.loc[:, :, self.fake_fov.streak_data.streak_channel]
+        )
+        self.fake_fov.streak_data.streak_mask = fake_img_bin_mask
+
+        sd._make_mask_dataframe(streak_data=self.fake_fov.streak_data, min_length=min_length)
```

Comment:
We should create the inputs for this function directly, rather than relying on another part of the pipeline to make the test data. This helps keep the tests more isolated from one another